Tests/Client/Unit Tests/Client+DevicesTests.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  Client+DevicesTests.swift
+//  StreamChatClientTests
+//
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import XCTest
+
+@testable import StreamChatClient
+
+class Client_DevicesTests: XCTestCase {
+
+    // Ideally, we would create a new client for every test case. Unfortunately, this is not technically
+    // possible at the moment.
+    lazy var client: Client = {
+        let sessionConfig = URLSessionConfiguration.default
+
+        sessionConfig.protocolClasses?.insert(RequestRecorderURLProtocol.self, at: 0)
+        Client.config = .init(apiKey: "test_api_key", defaultURLSessionConfiguration: sessionConfig)
+        return Client.shared
+    }()
+
+    func test_getDevice_createsRequest() {
+        // Setup
+        let testUser = User(id: "test_user_\(UUID())")
+        client.set(user: testUser, token: "test_token")
+
+        // Action
+        client.devices { _ in }
+
+        // Assert
+        AssertNetworkRequest(
+            method: .get,
+            path: "/devices",
+            headers: ["Content-Type": "application/json"],
+            queryParameters: ["api_key": "test_api_key"],
+            body: nil
+        )
+    }
+
+    func test_addDevice_withDeviceID_createsRequest() {
```

Comment:
Continuing from my comment above (about test naming) I suggest naming this test as:
`test_addDeviceWithDeviceId_createsRequest` since adding a device without a device ID is not possible.

Reply:
I agree with @b-onc

Reply:
fixed